File: src/main/java/com/ecom/pujakriti/entity/Bundle.java
package com.ecom.pujakriti.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Table(name="Bundle")
@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class Bundle {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "BundleID")
    private Integer bundleId;

    @Column(name = "Name", nullable = false)
    private String name;

    @Column(name = "Description")
    private String description;

    @Column(name = "Price", nullable = false)
    private Double price;

    @Column(name = "Stock", nullable = false)
    private Integer stock;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "GuideID")
    private Guide guide;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "PujaID")
    private Puja puja;
}

File: admin/src/types/Bundle.ts
import { Puja } from './Puja';
import { Guide } from './Guide';

export interface Bundle {
  bundleId: number;
  name: string;
  description?: string;
  price: number;
  stock: number;
  puja: string; 
  guide: string; 
}


export interface BundleDetails {
  bundleId: number;
  name: string;
  description?: string;
  price: number;
  stock: number;
  puja: number; 
  guide: number; 
}

export interface AddBundleRequest {
  name: string;
  description?: string;
  price: number;
  stock: number;
  puja: {
    name: string;
    description?: string;
  };
  guide: {
    name: string;
    description:string,
    content: string;
  };
}